### ISSUE TITLE
PC Kontrola nárokovaného stravného

### ISSUE THREAD

**Issue by Gubledon** [TIS2024-FMFI/gastro#57]
3.3.17. Nárokované stravné bude kontrolované voči časom začiatku a konca cesty, teda napr. ak začiatok cesty je nastavený na 17-tu hodinu, cestujúci si už nebude môcť nárokovať v ten deň o raňajky a obed

Relevantné súbory:

- **src\resources\views\business-trips\edit.blade.php** - pozrieť od riadku 379 a zamerať sa na 480 kde treba pridať validačnú logiku na základe $trip->datetime_start a $trip->datetime_end

**Comment by Gubledon** [comment]
Spravím pull request a commitnem tam niečo čo mi dáva časovo zmysel ale možno to budeme musieť upraviť. Myslím upraviť čisto časy čo bude veľmi rýchli a easy fix.

**Comment by Cerulikova** [comment]
Chcela som ísť pozrieť tento PR, ale keď si to už Lea začala, tak to nechám na teba a ja pozriem nejaký ďaľší.

**Comment by LeaMichalikova** [comment]
Počas stretnutia sme si vyjasnili situáciu, tým pádom už nemám žiadne dotazy. Mergujem to.

**Comment by Mcibula** [comment]
Ahojte, s tymto je problem, ze sa tie zmeny, ktore spravi tato logika, nezapisuju do DB, cize cesta, ktora ma takto zablokovane policka, ich v DB nema oznacene, a teda sa zda, ze tie jedla si pouzivatel narokuje preplatit, aj ked nema na to pravo.

Toto by bolo treba opravit, kedze takato tabulka sa po novom bude renderovat aj do dokumentov a co nie je v DB zapisane sa do dokumentu neprenesie.

Dakujem.

**Comment by Gubledon** [comment]
Pozriem sa na to

**Comment by Gubledon** [comment]
Vytvoril som pull request, čaká na schválenie.

**Comment by Cerulikova** [comment]
Ak to správne chápem, v databáze sa jedla ukladajú ako postupnosť 1 a 0, kde jednotky označujú nenárokované jedlá. Skúšala som vytvoriť cestu (má identifikátor 2222222), ktorá má automaticky začiarknuté jedlá v prvý deň raňajky a obed a v posledný deň obed a večeru. Okrem toho som ešte ja začiarkla ďaľšie dve jedlá (večera prvý deň a obed v predposledný deň).

Očakávala som teda, že v databáze bude postupnosť **11**10000100**11**. 
V skutočnosti sa však do databázy zapísalo len **00**10000100**00**.
A teda podľa DB to stále vyzerá, že tieto jedlá si cestujúci nárokuje. Prosím ešte opraviť.

**Comment by Cerulikova** [comment]
Ešte som si všimla jeden problém. Pokiaľ v prvý alebo posledný deň mám nárok na nejaké jedlá a označím, že si ich nenárokujem, všetko sa síce správne uloží do DB, ale po znovu-otvorení cesty sa tieto políčka zobrazia ako **nezačiarknuté**. To okrem vyzuálneho problému spôsobuje aj to, že ak si to používateľ nevšimne a políčka znovu neoznačí, prepíše sa po uložení akýchkoľvek zmien o ceste aj informácia o nárokovaných obedoch.

Čo som skúšala, stáva sa to iba **s prvým a posledným dňom cesty**, ostatné checkmárky sa zobrazujú správne.

**Comment by Gubledon** [comment]
Logiku som opravil a malo by to fungovať správne aj v DB, čaká na kontrolu a následný merge.

**Comment by Cerulikova** [comment]
No, do DB sa to už ukladá, ale... Teraz to nedovoluje klikať na checkboxy v celom prvom a poslednom dni. Čiže ak chcem ja zakliknúť, že okrem toho na čo nemám nárok si nenárokujem aj niečo iné, tak to nejde.

**Comment by Cerulikova** [comment]
Super, už to funguje
